app/lib/bk/compat/parsers/gha/matrix.rb

```diff
@@ -5,6 +5,7 @@ module Compat
     class GitHubActions
       def set_matrix(bk_step, config)
         matrix = config['strategy']['matrix']
+        matrix.map { |key, value| [key, value.map(&:to_s)] }.to_h
```

Comment:
`map` returns a copy of the array, you can ensure all values are strings with:

```suggestion
        matrix = config['strategy']['matrix'].transform_values(&:to_s)
```

Reply:
Oh damn that has changed the whole array to a string so its not quite what we want

Reply:
Oh... in that case I believe the solution would be have something like [the `recurse_to_string` method we use for transforming steps](https://github.com/buildkite/migration/blob/main/app/lib/bk/compat/pipeline/step.rb#L169-L181) applied here